flink-java/src/test/java/org/apache/flink/api/java/io/TextInputFormatTest.java

```diff
@@ -207,8 +206,110 @@ private void testRemovingTrailingCR(String lineBreaker, String delimiter) {
 				assertEquals(content, result);
 			}
 
+		} catch (Throwable t) {
+			System.err.println("test failed with exception: " + t.getMessage());
+			t.printStackTrace(System.err);
+			fail("Test erroneous");
+		}
+	}
+
+	@Test
+	public void testUTF16Read() {
```

Comment:
I think we need more tests, including tests that check correct behavior for LE and BE.